src/olympia/addons/management/commands/process_addons.py

```diff
@@ -160,6 +160,14 @@ def get_recalc_needed_filters():
         ],
         'allowed_kwargs': ('with_deleted',),
     },
+    'disable_opensearch_addons': {
+        # We're re-using the `delete_addons` method but don't allow for hard
+        # deletes
```

Comment:
Why don't we allow for hard deletes?  At some point in the next few months we will want to hard-delete all the opensearch addons we'll be soft deleting this time.  (And any soft-deleted opensearch add-ons in the database currently).

Reply:
It might be me being overly cautious but we may want to hard delete them at some point but not in the immediate future and we'll soft-delete first. That's why I separated them, not sure the cautiousness is appropriate.

Happy to change that if you think it isn't.

Reply:
I understand why we don't want to hard-delete them now, but I don't understand why you explicitly don't want the `--with_deleted` option available, so they can be hard deleted in the future without a further patch.